example/openMPSchedule/src/openMPSchedule.cpp

```diff
@@ -0,0 +1,163 @@
+/* Copyright 2019-2020 Benjamin Worpitz, Erik Zenker, Sergei Bastrakov
+ *
+ * This file exemplifies usage of alpaka.
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED “AS IS” AND ISC DISCLAIMS ALL WARRANTIES WITH
+ * REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL ISC BE LIABLE FOR ANY
+ * SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR
+ * IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#    include <alpaka/alpaka.hpp>
+#    include <alpaka/example/ExampleDefaultAcc.hpp>
+
+#    include <cstdint>
+#    include <iostream>
+
+//#############################################################################
+//! OpenMP schedule demonstration kernel
+//!
+//! Demonstrates how to set OpenMP schedule and prints distribution of alpaka
+//! thread indices between OpenMP threads
+struct OpenMPScheduleKernel
+{
+    // A simple way to set a schedule for all invocations on this kernel using a constexpr member.
+    // Note that constexpr is not required, however otherwise there has to be an external definition.
+    // Another way to set the schedule is via trait specialization, as demonstrated after the kernel body.
+    // Trait specialization is checked first, so the member only has effect when the trait is not specialized for this
+    // type.
+    static constexpr auto ompSchedule = alpaka::omp::Schedule{alpaka::omp::Schedule::Static, 1};
+
+    //-----------------------------------------------------------------------------
+    template<typename TAcc>
+    ALPAKA_FN_ACC auto operator()(TAcc const& acc) const -> void
+    {
+        // For simplicity assume 1d index space throughout this example
+        using Idx = alpaka::Idx<TAcc>;
+        Idx const globalThreadIdx = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0];
+
+        // Print work distribution between threads for illustration
+        printf(
+            "alpaka global thread index %u is processed by OpenMP thread %d\n",
+            static_cast<std::uint32_t>(globalThreadIdx),
+            omp_get_thread_num());
```

Comment:
OSX bug: If OpenMP is not available e.g. if the OMP backend is not enabled than you can not call a native OpenMP function. 
This print should be guarded with `#if`

Reply:
Ah, yes. I originally guarded the whole thing but then changed it. Will change back